code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -0,0 +1,495 @@
+	for(var/obj/structure/telecomms_trap_tank/TTT in urange(15, get_turf(src)))
+		TTT.explode()
+	flick_holder.layer = kaboom.layer + 0.1
+	flick("popup", flick_holder)
+	sleep(1 SECONDS)
+	for(var/obj/machinery/shieldgen/telecomms/shield in urange(15, get_turf(src)))
+		shield.shields_up()
+	if(ruin_cheese_attempted)
+		for(var/obj/machinery/door/airlock/A in urange(20, get_turf(src)))
+			A.unlock(TRUE) //Fuck your bolted open doors, you cheesed it.
+			A.close(override = TRUE)
+	for(var/area/A in urange(25, get_turf(src), areas = TRUE))
+		if(istype(A, /area/space))
+			continue
+		if(ruin_cheese_attempted)
+			A.burglaralert(src)
+		else if(!A.fire)
+			A.firealert(kaboom)
+	for(var/obj/effect/abstract/cheese_trap/CT in urange(15, get_turf(src)))
+		qdel(CT)
+	kaboom.activate()
+	kaboom.icon_state = "death-bomb-active" // something funny goes on with icons here
+	qdel(flick_holder)
+	qdel(src)
+
+/obj/machinery/syndicatebomb/doomsday
+	name = "\improper D.V.O.R.A.K's Doomsday Device"
+	icon_state = "death-bomb"
+	desc = "Nice to see that AI's are improving on the standard doomsday device. Good to have variety. Also probably a good idea to start running."
+	anchored = TRUE
+	timer_set = 100
+	payload = /obj/item/bombcore/doomsday
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+
+/obj/machinery/syndicatebomb/doomsday/singularity_act()
+	return // saves me headache later
+
+/obj/machinery/syndicatebomb/doomsday/ex_act(severity)
+	return // No.
+
+/obj/machinery/syndicatebomb/doomsday/screwdriver_act(mob/user, obj/item/I)
+	to_chat(user, "<span class='danger'>[src] is welded shut! You can't get at the wires!</span>")
+
+/obj/machinery/syndicatebomb/doomsday/Initialize(mapload)
+	. = ..()
+	var/turf/T = get_turf(src)
+	if(is_station_level(T.z))
+		log_debug("something tried to spawn a telecomms doomsday ruin bomb on the station, deleting!")
+		return INITIALIZE_HINT_QDEL
+
+/obj/item/bombcore/doomsday
+	name = "a supermatter charged bomb core"
+	desc = "If you are looking at this, please don't put it in a bomb"
+
+/obj/item/bombcore/doomsday/Initialize()
+	. = ..()
+	if(!istype(loc, /obj/machinery/syndicatebomb/doomsday))
+		log_debug("something tried to spawn a telecomms doomsday ruin payload outside the ruin, deleting!")
+		return INITIALIZE_HINT_QDEL
+
+/obj/item/bombcore/doomsday/ex_act(severity) // No
+	return
+
+/obj/item/bombcore/doomsday/burn() // Still no.
+	return
+
+/obj/item/bombcore/doomsday/detonate()
+	if(loc && istype(loc, /obj/machinery/syndicatebomb))
+		loc.invisibility = 90
+	for(var/turf/simulated/wall/indestructible/riveted/R in urange(25, get_turf(src)))
+		R.ChangeTurf(/turf/space)
+	explosion(get_turf(src), 30, 40, 50, 60, 1, 1, 65, 0)
+	sleep(3 SECONDS)
+	var/obj/singularity/S = new /obj/singularity(get_turf(src))
+	S.consumedSupermatter = TRUE // woe large supermatter to eat the remains apon thee
+	S.energy = 4000
+	sleep(25 SECONDS)
```

Comment:
Can you do this with a timer?

Reply:
probably yeah

Reply:
OOH, qdel_in, perfect

Reply:
Will use qdel_in, the only people that would see the visable message is ghosts that stuck around, so not an issue